### ISSUE TITLE
5 high severity vulnerabilities

### ISSUE THREAD

**Issue by NewMountain** [sanack/node-jq#410]
Hello sanack and the maintainers of node-jq. My apologies if you are already aware of the issue and please feel free to close if this is not useful. I did not see such an issue open.

When installing node-jq, I'm getting a rather scary error

```
❯ npm install --save node-jq
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/pinpoint@1.0.2: Moved to 'npm install @sideway/pinpoint'
npm WARN deprecated @hapi/formula@1.2.0: Moved to 'npm install @sideway/formula'
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@16.1.8: Switch to 'npm install joi'

added 232 packages, and audited 233 packages in 7s

42 packages are looking for funding
  run `npm fund` for details

5 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

In running npm audit, I get:

```
❯ npm audit                 
# npm audit report

normalize-url  <=4.5.0 || 5.0.0 - 5.3.0 || 6.0.0
Severity: high
Regular Expression Denial of Service - https://npmjs.com/advisories/1755
fix available via `npm audit fix --force`
Will install node-jq@1.3.0, which is a breaking change
node_modules/normalize-url
  cacheable-request  0.1.0 - 6.0.0
  Depends on vulnerable versions of normalize-url
  node_modules/cacheable-request
    got  8.0.0 - 9.5.0
    Depends on vulnerable versions of cacheable-request
    node_modules/got
      download  >=7.0.0
      Depends on vulnerable versions of got
      node_modules/download
        node-jq  >=1.3.1
        Depends on vulnerable versions of download
        node_modules/node-jq

5 high severity vulnerabilities
```

In digging through a bit, I see a few issues filed on [`kevva/download`](https://github.com/kevva/download/issues/216) and a fix PR in `leafac/caxa` from a [similar error](https://github.com/leafac/caxa/pull/32)

I believe this is related to a dependency of node-jq and not related to a vulnerability in jq itself.

## Environment

Tell us which operating system you are using, as well as which versions of Node.js, npm/yarn, and node-jq.

I'm running

```
Node.js v14.15.1
darwin 20.5.0
yarn 1.22.10
npm 7.18.1
node-jq 1.12.0
```

In digging through a bit more, there was some discussion [here](https://github.com/sindresorhus/normalize-url/issues/135) that all of this may originate from a spurious CVE.

Hopefully this was useful and thank you again for your collective effort!

**Comment by davesnx** [comment]
Hey @NewMountain 

There are a few packages related with `hapi` that are deprecated that we rely on, the rest of the packages as you pointed out happen to be on https://github.com/kevva/download, once this gets resolved it's a tiny dependency update on our side.

Thanks for opening the issue

**Comment by davesnx** [comment]
Hey @NewMountain 

I got the time to fix the issues you mentioned above. The only warning that I got when installing is from `uuid@3.3.2`. Which happens to be on the transitive dependencies from node-jq, so there are a few versions of `bin-build`, `tempfile` and even `codeclimate-test-reporter` so it's unlikely to get removed soon.

```
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
```

Thanks for caring and opening this issue <3
